Add support for JSCS and wordpress preset

westonruter opened the pull request:
Fixes #38 

Install JSCS:

```bash
npm install jscs -g
```

PhpStorm configuration:

![phpstorm-config](https://cloud.githubusercontent.com/assets/134745/6931255/c9e28082-d7bf-11e4-95ea-b97925255f79.png)

hzoo commented:
You can also do `jscs --preset=wordpress, but I think the `.jscsrc` file is better if any rules need to be added later/remove without having to wait a new jscs release.

`--verbose` is useful because it prints out the violating **rule name** as well.

If you need to exclude files - in the `.jscsrc`
```js
"excludeFiles": ["node_modules/**", "src/!(bar|foo)"]
```

westonruter commented:
@hzoo thanks! 

I added `excludeFiles` to correspond to what we have in [`.jshintignore`](https://github.com/xwp/wp-dev-lib/blob/master/.jshintignore): https://github.com/xwp/wp-dev-lib/commit/bbd7716d883cb20eb9dd6a345810339e516cbeaa

And then I added `--verbose` to where `jscs` is called in the `pre-commit` hook and Travis script: https://github.com/xwp/wp-dev-lib/commit/c5f2447c2d82b86ac16548e100f98b7d4605ec81

hzoo commented:
Oh yeah - one more thing in 1.12 is an autofix option for whitespace rules (`jscs --fix`) that you can run which helps a lot with getting the codebase up to speed initially.
